webapp/channels/src/packages/mattermost-redux/src/reducers/entities/scheduled_posts.ts

```diff
@@ -178,23 +178,22 @@ function byChannelOrThreadId(state: ScheduledPostsState['byChannelOrThreadId'] =
         const {scheduledPostsByTeamId} = action.data;
         const newState = {...state};
 
+        const newIdsByChannel: {[key: string]: Set<string>} = {};
+
         Object.keys(scheduledPostsByTeamId).forEach((teamId: string) => {
-            if (Object.hasOwn(scheduledPostsByTeamId, teamId)) {
-                scheduledPostsByTeamId[teamId].forEach((scheduledPost: ScheduledPost) => {
-                    const id = scheduledPost.root_id || scheduledPost.channel_id;
-
-                    // Check if the entry for that channel/thread ID exists
-                    if (newState[id]) {
-                        // Only add if its not already there
-                        if (!newState[id].includes(scheduledPost.id)) {
-                            newState[id] = [...newState[id], scheduledPost.id];
-                        }
-                    } else {
-                        // If the entry does not exist at this moment, create it
-                        newState[id] = [scheduledPost.id];
-                    }
-                });
-            }
+            scheduledPostsByTeamId[teamId].forEach((scheduledPost: ScheduledPost) => {
+                const channelOrThreadId = scheduledPost.root_id || scheduledPost.channel_id;
+                if (!newIdsByChannel[channelOrThreadId]) {
+                    newIdsByChannel[channelOrThreadId] = new Set();
+                }
+                newIdsByChannel[channelOrThreadId].add(scheduledPost.id);
+            });
+        });
+
+        Object.keys(newIdsByChannel).forEach((channelOrThreadId) => {
+            const existingSet = new Set(newState[channelOrThreadId] || []);
```

Comment:
existing IDs are merged with the new set, making sure that the duplicates are removed